dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/LoggerUtils.java

```diff
@@ -71,12 +72,14 @@ private LoggerUtils() {
      * @return task id format
      */
     public static String buildTaskId(String affix,
+                                     Date firstSubmitTime,
                                      Long processDefineCode,
                                      int processDefineVersion,
                                      int processInstId,
                                      int taskId) {
-        // - [taskAppId=TASK-798_1-4084-15210]
-        return String.format(" - %s%s-%s_%s-%s-%s]", TASK_APPID_LOG_FORMAT, affix, processDefineCode, processDefineVersion, processInstId, taskId);
+        // - [taskAppId=TASK-20211107-798_1-4084-15210]
+        String firstSubmitTimeStr = DateUtils.format(firstSubmitTime, Constants.YYYYMMDD);
+        return String.format(" - %s%s-%s-%s_%s-%s-%s]", TASK_APPID_LOG_FORMAT, affix, firstSubmitTimeStr, processDefineCode, processDefineVersion, processInstId, taskId);
```

Comment:
LGTM, 
but why not use all `-` here? Use `_` here is just want to split?

Reply:
Maybe we could found it git history to check out why

Reply:
Before org.apache.dolphinscheduler.server.log.TaskLogDiscriminator#getDiscriminatingValue replace `-` with `/`

Reply:
I mean why not use a consistent separator?  The current log looks like `453920594116992_1-1-1.log`, I prefer `453920594116992-1-1-1.log`. This maybe a history issue.
Anyway, it is not a big deal, maybe I like to use consistent separator.

It's better to extract code from `LogUtils` and `LoggerUtils`, otherwise it may easy to generate bug if someone change the path.
It can be a future job.

Reply:
> I mean why not use a consistent separator? The current log looks like `453920594116992_1-1-1.log`, I prefer `453920594116992-1-1-1.log`. This maybe a history issue. Anyway, it is not a big deal, maybe I like to use consistent separator.
> 
> It's better to extract code from `LogUtils` and `LoggerUtils`, otherwise it may easy to generate bug if someone change the path. It can be a future job.

I agree,i am confused about `LogUtils` and `LoggerUtils`  when i do this feature.

Reply:
> I agree,i am confused about `LogUtils` and `LoggerUtils` when i do this feature.

@choucmei Do you want to submit a PR to fix this confused thinks?